dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi

```diff
@@ -70,10 +58,18 @@
 	 * I2C		<0xffc02800 0x200>
 	 * I2C EMAC	<0xffc02a00 0x300>
 	 * Timer SP	<0xffc03000 0x200>
+	 * GPIO_0	<0xffc03200 0x100>
+	 * GPIO_1	<0xffc03300 0x100>
+	 * Rst Manager	<0xffd11000 0x1000>
+	 * Sys Manager	<0xffd12000 0x1000>
+	 * Pinmux	<0xffd13000 0x1000>
```

Comment:
again, this is abusing the syscon device. If (for example) the pinmux driver needs to MMU map some region, it's the pinmux driver that needs to do that.

Reply:
Hi Carlo, we are yet to enable a pinmux driver. However, in the upcoming pull request on the GPIO driver, there is a need to read from the pinmux address register (not to set), so, we map the pinmux register using the syscon device.

Reply:
I see the issue. Let me talk with @npitre, we probably want to move the assert when the page has been already mapped.

Reply:
Hi Carlo, 

Let us know the ETA when the assert would be moved. Thanks

Reply:
There is no ETA, I'll open an issue to track this but if this is a blocker for you I strongly suggest that someone from Intel would propose a PR to fix the MMU code. I'll try to see if I can find time to look at this soon-ish.

Reply:
Okay,  thanks a lot Carlo. 

Help me to take a look on the comment on the previous line of code as well, regarding the mmu mapping on driver, need your suggestion on that. Thanks :)